src/lib/fauna-client.mjs

```diff
@@ -71,7 +71,17 @@ export default class FaunaClient {
   }
 }
 
-export const getSecret = async () => {
+/**
+ * Gets a secret. Defaults to the secretArg. If the secretArg is undefined
+ * fetches a secret for the current credentials.
+ * @param opts {Object}
+ * @param opts.secretArg {string | undefined} The secret passed as an argument, if any.
+ * @return {Promise<string>} the secret
+ */
+export async function getSecret({ secretArg }) {
+  if (secretArg !== undefined) {
+    return secretArg;
+  }
```

Comment:
Credential related arguments should be settled in the middleware so we can have all the various sources and defaults resolved in one place. I am thinking here:
https://github.com/fauna/fauna-shell/blob/v3/src/lib/auth/databaseKeys.mjs#L47-L61

have:
```
if (argv.secret) {
    key = argv.secret;
    keySource = "user";
  }  else if (argv.url) {
    key = "secret";
    keySource = "local-default"
 } else
...
```

after that this `getSecret` function should find "secret" in `credentials.databaseKeys.key`

Reply:
OK seems like the middleware has a bug then. I'll fix it.